### PR TITLE
Add batch decomposition for torch.unsafe_chunk

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110862

This updates the docs as well to show `torch.unsafe_chunk`. Should the `unsafe_*` functions should not appear in the docs?

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -10425,6 +10425,14 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
            supports_out=False),
+    OpInfo('unsafe_chunk',
+           dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16, torch.float16, torch.chalf),
+           sample_inputs_func=sample_inputs_chunk,
+           check_batched_forward_grad=False,
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:10431]: gradcheck uses legacy vmap implementation for batched forward and `unsafe_chunk` uses `unsafe_split` which doesn't have a batching rule for legacy vmap. (`unsafe_split` OpInfo also sets this to False)

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks @guilhermeleobas 

cc: @zou3519 for binding approval.

File: docs/source/torch.rst [hunk 2]
```diff
@@ -133,6 +133,7 @@ Indexing, Slicing, Joining, Mutating Ops
     transpose
     unbind
     unsqueeze
+    unsafe_chunk
```
Comment (zou3519) [inline docs/source/torch.rst:136]: The unsafe_* APIs should not show up in the docs. We don't want users using them directly.

**Review by zou3519 — COMMENTED** [review]
Please remove the mention from the docs, otherwise, this LGTM

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**